LJ-528 simplify container image and local development

nick-wilson-lji opened the pull request.

BradWells commented on build/base/Dockerfile:
```diff
@@ -1,60 +0,0 @@
-FROM python:3.9-slim
```
cron/Dockerfile also depends on the base image.
@amunn33 Do we use that at all? I think we were just using crons on Linode directly and not any docker container so I think that could be deleted, too.

I also don't think we are using deploy/Dockerfile either since it is just sshing and we are doing that with commands on the github action instead.

BradWells replied:
Also, I would search the codebase for dreadsuite_tag and find out where it is used. Our build process was to store the version number of what needs to be pulled from the docker image store and without a base image that whole process should be ravamped or removed, I think.

BradWells commented on pyproject.toml:
```diff
@@ -0,0 +1,71 @@
+[project]
```
Changing my Pycharm Tools -> Python Integrated Tools -> Package requirements file to pyproject.toml "just worked" so I'm ok with this change, I assume that Cursor and similar tooling can handle this.

BradWells commented on scripts/build-containers.sh:
.github/workflows/build_test_deploy.yml needs to have path changed

BradWells commented on scripts/clone_file_backups_to_azure.sh:
prod-crontab.txt needs to be changed

BradWells commented on scripts/push-containers.sh:
Will we need this?

BradWells commented on scripts/send_backups_to_azure.sh:
prod-crontab.txt needs to be changed

BradWells commented on application/dreamsuite/settings/base.py:
```diff
@@ -5,6 +5,18 @@
+from dotenv import load_dotenv
+
+# Load environment variables from a .env file if it exists
+BASE_DIR = Path(__file__).ancestor(4)  # Reference the project root
```
Rename this as project dir and django base dir? There are two BASE_DIR variables.

BradWells commented on application/dreamsuite/settings/base.py:
```diff
@@ -5,6 +5,18 @@
+
+# Load environment variables from a .env file if it exists
+BASE_DIR = Path(__file__).ancestor(4)  # Reference the project root
+common_env_path = BASE_DIR.child('common.env')
```
I'm not sold on doing it this way yet.

BradWells replied:
We talked about this and for now we will move these load_dotenv() commands to scripts.py in order to make them work locally without a DotEnv plugin in your IDE. Then this would only change how local development works rather than also changing how staging and prod work.

In the future, we could have the contents of different .env files be in Github Secrets to be sent to different servers, but changing the way we handle Docker environments touches a lot of different things and I'm not ready to bite that off right now.

BradWells commented on pyproject.toml:
```diff
@@ -0,0 +1,71 @@
+dev = "application.scripts:dev"
+migrate = "application.scripts:migrate"
+test = "application.scripts:test"
+shell = "application.scripts:shell"
```
In addition to these, I'd want a "django" command that is flexible and takes arguments from the command line and a "djangoHelp" command that prints the help function that lists all the commands you can do.

nick-wilson-lji commented:
@BradWells I removed the commit that moved the scripts to /scripts. I also added a `manage_local.py` for local development and removed the commands from `pyproject.toml`.

BradWells reviewed (approved):
Approved, but check the files I sent you to make sure prod deploy works
